play.pokemonshowdown.com/src/panels.tsx

```diff
@@ -349,6 +349,37 @@ export class PSView extends preact.Component {
 			return PS.prefs.refreshprompt ? "Are you sure you want to leave?" : null;
 		};
 
+		window.addEventListener('submit', ev => {
+			let elem = ev.target as HTMLFormElement | null;
+			if (elem?.getAttribute('data-submitsend')) {
+				const elements = elem.querySelectorAll('input[name], select[name], textarea[name], keygen[name], button[value]');
```

Comment:
`keygen` does not appear to be a real HTML element? What is it doing here?

Reply:
Looks like it's a proposed tag that's considered obsolete. Probably we shouldn't use it.

Reply:
It was removed in Firefox 69 for instance. (I didn't check other browsers.)